cdap-examples/LogAnalysis/src/main/java/co/cask/cdap/examples/loganalysis/LogAnalysisApp.java

```diff
@@ -0,0 +1,165 @@
+        .join();
+    }
+  }
+
+  /**
+   * Specification for the Spark program in this application
+   */
+  public static final class SparkKMeansSpecification extends AbstractSpark {
+
+    @Override
+    public void configure() {
+      setName(ResponseCounterProgram.class.getSimpleName());
+      setDescription("Counts the total number of responses for every unique response code");
+      setMainClass(ResponseCounterProgram.class);
+    }
+  }
+
+  /**
+   * A {@link Service} that responds with total number of hits for a given URL or path
+   */
+  public static final class HitCounterServiceHandler extends AbstractHttpServiceHandler {
+
+    private static final Gson GSON = new Gson();
+    public static final String URL_KEY = "url";
+    public static final String HIT_COUNTER_SERVICE_PATH = "hitcount";
+
+    @UseDataSet(HIT_COUNT_STORE)
+    private KeyValueTable hitCountStore;
+
+    @Path(HIT_COUNTER_SERVICE_PATH)
+    @POST
+    public void getHitCount(HttpServiceRequest request, HttpServiceResponder responder) {
+      String urlRequest = Charsets.UTF_8.decode(request.getContent()).toString();
+      String url = GSON.fromJson(urlRequest, JsonObject.class).get(URL_KEY).getAsString();
+      if (url == null) {
+        responder.sendString(HttpURLConnection.HTTP_BAD_REQUEST,
+                             "A url or path must be specified with \"url\" as key in JSON.",
+                             Charsets.UTF_8);
+        return;
+      }
+
+      // Get the total number of hits from the dataset for this path
+      byte[] hitCount = hitCountStore.read(url.getBytes(Charsets.UTF_8));
+      if (hitCount == null) {
+        responder.sendString(HttpURLConnection.HTTP_NO_CONTENT,
+                             String.format("No record found of %s", url), Charsets.UTF_8);
+      } else {
+        responder.sendString(HttpURLConnection.HTTP_OK, String.valueOf(Bytes.toInt(hitCount)), Charsets.UTF_8);
+      }
+    }
+  }
+
+  /**
+   * A {@link Service} that responds with total number of responses for a given response code
+   */
+  public static final class ResponseCounterHandler extends AbstractHttpServiceHandler {
+
+    public static final String RESPONSE_COUNT_PATH = "rescount";
+
+    @UseDataSet(RESPONSE_COUNT_STORE)
+    private KeyValueTable responseCountstore;
+
+    @Path(RESPONSE_COUNT_PATH + "/{rescode}")
+    @GET
+    public void centers(HttpServiceRequest request, HttpServiceResponder responder,
+                        @PathParam("rescode") Integer responseCode) {
+
+      byte[] read = responseCountstore.read(Bytes.toBytes(responseCode));
+      if (read == null) {
+        responder.sendString(HttpURLConnection.HTTP_NO_CONTENT,
+                             String.format("No record found for response code: %s", responseCode), Charsets.UTF_8);
+      } else {
+        responder.sendString(HttpURLConnection.HTTP_OK, String.valueOf(Bytes.toLong(read)), Charsets.UTF_8);
```

Comment:
You can just use `sendString(String)`. Not sure if he wants to do Json response.